pkg/parquet/reader.go

```diff
@@ -197,6 +220,23 @@ func (r *s3Reader) ReadFile(ctx context.Context, file string) (ReadResults, erro
 	}
 
 	size := int(pr.GetNumRows())
+
+	if batchSize != -1 {
+		err = pr.SkipRows(offset)
+
+		if err != nil {
+			return nil, err
+		}
+
+		leftRows := size - int(offset)
+
+		size = batchSize
+
+		if batchSize > leftRows {
+			size = leftRows
+		}
+	}
```

Comment:
```suggestion

	if offset > 0 {
		err = pr.SkipRows(offset)
		if err != nil {
			return nil, err
		}
	}

	if batchSize > 0 {
		remainingRows := size - int(offset)

		size = batchSize

		if batchSize > remainingRows {
			size = remainingRows
		}
	}
```

Reply:
i can't add a suggestion below changes code, but i'd also move `results := make(ReadResults, size)` up after this block and return it `if size == 0`